### PR TITLE
[ISSUE #5322] improving SDK topic route availability

### PR DESCRIPTION
**Pull request by fuyou001**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: common/src/main/java/org/apache/rocketmq/common/namesrv/NamesrvConfig.java [hunk 1]
```diff
@@ -78,6 +78,9 @@ public class NamesrvConfig {
      */
     private boolean enableControllerInNamesrv = false;
 
+    private volatile boolean needWaitForService = true;
+
+    private int waitSecondsForService = 45;
```
Comment (lizhanhui) [inline common/src/main/java/org/apache/rocketmq/common/namesrv/NamesrvConfig.java:83]: Hard-coded value is not optimal, users might adjust the interval of broker registration... Would it be better to set it to alpha * interval_of_broker_registration, where alpha is somewhat 1.5 or 2?

**Comment by lizhanhui** [comment]
@fuyou001 A few test cases failed after this change...Please look into them

File: common/src/main/java/org/apache/rocketmq/common/namesrv/NamesrvConfig.java (hunk 1 above)
Comment (fuyou001) [reply common/src/main/java/org/apache/rocketmq/common/namesrv/NamesrvConfig.java:83]: broker config and namesrv config is independence，it's hard to get broker interval_of_broker_registration in namesrv.

**Comment by fuyou001** [comment]
> @fuyou001 A few test cases failed after this change...Please look into them
fix it

**Review by lizhanhui — APPROVED** [review]

**Pull request merged**